pyg_lib/csrc/sampler/cpu/indices_tracker.h

```diff
@@ -0,0 +1,49 @@
+#pragma once
+
+#include <ATen/ATen.h>
+
+#include "parallel_hashmap/phmap.h"
+
+namespace pyg {
+namespace sampler {
+
+template <typename scalar_t>
+class IndicesTracker {
+ public:
+  IndicesTracker(const scalar_t& size) : size(size) {
+    // TODO: add better switching threshold value mechanism?
+    use_vec = (size < 100000);
```

Comment:
I feel this population threshold is a bit too large. I feel we can start with something like 100 or 1000 and see how it goes.

Reply:
I was doing some init analyses of using vector<char> vs unorder_set on neighborloader bench in pyg for ogbn-products and vector was always faster were max population for products was 10000+. Alternatively tested also vector<char> with using byte as storage for  bits and vector<bool> - vector<bool> was the slowest. Vector<char> as 8bit storage was also a little bit slower than vector<char>. In case of trade-off between memory usage vs perf, We can think about what to apply and what threshold to use.

Reply:
Yes, main point I wanted to make is if we shouldn't just use `std::vector` in the first place in the sampling without replacement logic. There is likely never a case where the population exceeds 10,000.

Reply:
"bench in pyg for ogbn-products and vector was always faster where max population for products was 10000+" -sorry, that sentence in misleading. I wanted to say that base on some initial research: vector was  faster than set for any population. And in case of products the populations max was going up to 10000+.